debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -511,7 +511,7 @@ public synchronized void close() {
      */
     public Long currentTransactionId() throws SQLException {
         AtomicLong txId = new AtomicLong(0);
-        query("select * from txid_current()", rs -> {
+        query("select (case pg_is_in_recovery() when 't' then 0 else txid_current() end) AS pg_current_txid", rs -> {
```

Comment:
I think returning `null` in the stand-by case from this method is fine, the tx id is only used for contextual info anyways. Note this only makes a difference upon start-up, actual change events do have the tx id as per the transaction from the primary.